rpcs3/evdev_joystick_handler.cpp

```diff
@@ -251,98 +373,117 @@ std::vector<std::string> evdev_joystick_handler::ListDevices()
     return evdev_joystick_list;
 }
 
+int evdev_joystick_handler::add_device(const std::string& device, std::shared_ptr<Pad> pad)
+{
+	// Now we need to find the device with the same name, and make sure not to grab any duplicates.
+	fs::dir devdir{ "/dev/input/" };
+	fs::dir_entry et;
+	while (devdir.read(et))
+	{
+		// Check if the entry starts with event (a 5-letter word)
+		if (et.name.size() > 5 && et.name.compare(0, 5, "event") == 0)
+		{
+			std::string path = "";
+			int fd = open(("/dev/input/" + et.name).c_str(), O_RDONLY | O_NONBLOCK);
+			struct libevdev *dev = NULL;
+			int rc = libevdev_new_from_fd(fd, &dev);
+			if (rc < 0)
+			{
+				// If it's just a bad file descriptor, don't bother logging, but otherwise, log it.
+				if (rc != -9)
+					LOG_WARNING(GENERAL, "Failed to connect to device at %s, the error was: %s", "/dev/input/" + et.name, strerror(-rc));
+				libevdev_free(dev);
+				close(fd);
+				continue;
+			}
+			const std::string name = libevdev_get_name(dev);
+			if (libevdev_has_event_type(dev, EV_KEY) &&
+				libevdev_has_event_code(dev, EV_ABS, ABS_X) &&
+				libevdev_has_event_code(dev, EV_ABS, ABS_Y) &&
+				name == device)
+			{
+				// It's a joystick.
+
+				// Now let's make sure we don't already have this one.
+				auto it = std::find_if(devices.begin(), devices.end(), [path](EvdevDevice device) { return path == device.path; });
```

Comment:
shouldn't it be `[&path](const EvdevDevice &device)`. No reason to copy strings around.